packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -482,9 +482,15 @@ gmaps.PolygonOptions _polygonOptionsFromPolygon(
   polygon.points.forEach((point) {
     path.add(_latLngToGmLatLng(point));
   });
+  final polygonDirection = _isPolygonClockwise(path);
   List<List<gmaps.LatLng>> paths = [path];
   polygon.holes?.forEach((hole) {
-    paths.add(hole.map((point) => _latLngToGmLatLng(point)).toList());
+    final holePath = hole.map((point) => _latLngToGmLatLng(point)).toList();
+    if (_isPolygonClockwise(holePath) == polygonDirection) {
+      print(
+          'Holepath is the same direction as polygon. In order to be displayed properly you should reverse the hole Path. More Info: https://developers.google.com/maps/documentation/javascript/examples/polygon-hole');
+    }
```

Comment:
You might be able to replace this by an `assert(_isPolygonClockwise(holePath) == _isPolygonClockwise(path), 'The path of the hole should be reversed. See: blah');`. IIRC this is catchable in tests and will nag users (in development mode) for them to use the correct polygon winding.

Another thing we can do is just `.revert` the holePath before using it, and log the error. Then in the test you can just verify that the hole has been automatically reversed. I kind of like this second idea better, this brings the web to feature parity with the mobile version!

Reply:
I changed it so the holepath gets reversed if the direction matches and added a test for it.
Also reverted the example to the original state